firedrake/slate/slate.py

```diff
@@ -625,13 +627,12 @@ def merge_coefficients(operands, assembled_coeffs=None):
     :arg assembled_coeffs: a `tuple` of assembled data provided
                            as a `firedrake.Function`.
     """
-    coefficient_list = []
-    extra_data = assembled_coeffs or ()
-    for op in operands:
-        coefficient_list.extend(op.coefficients())
+    if assembled_coeffs is None:
+        coeffs = [op.coefficients() for op in operands]
+    else:
+        coeffs = [op.coefficients() for op in operands] + [assembled_coeffs]
```

Comment:
I think you could just inline this function now, if I counted correctly, it is only called once with `assembled_coeffs` and once without it.